File: app/src/main/java/include/mvp/data/module/MainScreenModule.java
package include.mvp.data.module;

import dagger.Module;
import dagger.Provides;
import include.mvp.mainscreen.MainScreenContract;
import include.mvp.util.CustomScope;

/**
 * Created by iagomendesfucolo on 26/12/16.
 */

@Module
public class MainScreenModule {
    private final MainScreenContract.View mView;


    public MainScreenModule(MainScreenContract.View mView) {
        this.mView = mView;
    }

    @Provides
    @CustomScope
    MainScreenContract.View providesMainScreenContractView() {
        return mView;
    }
}

File: app/src/main/java/include/mvp/data/module/MainScreenModule.kt
package include.mvp.data.module

import javax.inject.Singleton

import dagger.Module
import dagger.Provides
import include.mvp.mainscreen.MainActivity
import include.mvp.mainscreen.MainScreenContract
import include.mvp.presenter.MainPresenter

/**
 * Created by iagomendesfucolo on 26/12/16.
 */

@Module
class MainScreenModule {


    @Singleton
    @Provides
    fun providesPresenter(presenter: MainPresenter): MainScreenContract.Presenter {
        return presenter
    }
}
